ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/catalog.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from yaml import safe_load
+
+from ansible_collections.arista.avd.plugins.plugin_utils.errors import AristaAvdError
+from ansible_collections.arista.avd.roles.eos_validate_state.python_modules.constants import AVD_TEST_CLASSES
+
+from .get_eos_validate_state_vars import get_eos_validate_state_vars
+
+if TYPE_CHECKING:
+    from .avdtestbase import AvdTestBase
+
+
+class Catalog:
+    """
+    Catalog class that handles management of ANTA catalogs from AVD.
+    """
+
+    def __init__(self, device_name: str, hostvars: dict, skipped_tests: dict, custom_catalog: dict | None = None, custom_catalog_path: str | None = None):
+        """
+        Initialize the Catalog class.
+
+        For now it generates a catalog specific to a given device
+
+        Args:
+            device_name (str): The device for which this catalog is built
+            hostvars (dict): A dictionnary that contains a key for each device with a value of the structured_config
+                         when using Ansible, this is the equivalent of `task_vars['hostvars']`
+            skipped_tests (dict): Dictionary of AVD test classes to be skipped.
+            custom_catalog (dict): An optional dictionary representing an ANTA Catalog
+            custom_catalog_path (str): An optional string representing a path to an Anta Catalog
+        """
+        self.device_name = device_name
+        self.hostvars = hostvars
+        self.skipped_tests = skipped_tests
+
+        self.catalog = self._input_catalog(custom_catalog, custom_catalog_path) or self._default_catalog()
+
+    def _input_catalog(self, custom_catalog, custom_catalog_path) -> dict:
+        """
+        NOTE: Not being used by the anta_runner plugin yet. Only the default_catalog is used.
```

Comment:
this note says it is not being used, but 3 lines above it is called in __init__. What do we want?

Reply:
it is used in `init` but you cannot pass these argument to the function from ansible_module

Reply:
I guess the issue here is that I renamed anta_runner in eos_validate_state_runner and so this comment is very obscure